widgetastic_manageiq.py

```diff
@@ -320,6 +321,62 @@ def click_path(self, *path, **kwargs):
         return leaf
 
 
+class CheckableDynaTree(DynaTree):
+    """ Checkable variation of CFME Tree. This widget not only expand a tree for a provided path,
+    but also checks a checkbox.
+    """
+
+    IS_CHECKABLE = './../span[contains(@class, "dynatree-checkbox")]'
+    IS_CHECKED = './../../span[contains(@class, "dynatree-selected")]'
+
+    def is_checkable(self, item):
+        return bool(self.browser.elements(self.IS_CHECKABLE, parent=item))
+
+    def is_checked(self, item):
+        return bool(self.browser.elements(self.IS_CHECKED, parent=item))
+
+    def check_uncheck_node(self, check, *path, **kwargs):
+        leaf = self.expand_path(*path, **kwargs)
+        if not self.is_checkable(leaf):
+            raise TypeError('Item is not checkable')
+        checked = self.is_checked(leaf)
+        if checked != check:
+            self.logger.info('%s %r', 'Checking' if check else 'Unchecking', path[-1])
+            self.browser.click(self.IS_CHECKABLE, parent=leaf)
+
+    def check_node(self, *path, **kwargs):
+        """Expands the passed path and checks a checkbox that is located at the node."""
+        return self.check_uncheck_node(True, *path, **kwargs)
+
+    def uncheck_node(self, *path, **kwargs):
+        """Expands the passed path and unchecks a checkbox that is located at the node."""
+        return self.check_uncheck_node(False, *path, **kwargs)
+
+    def node_checked(self, *path, **kwargs):
+        """Check if a checkbox is checked on the node in that path."""
+        leaf = self.expand_path(*path, **kwargs)
+        if not self.is_checkable(leaf):
+            return False
+        return self.is_checked(leaf)
+
+    def fill(self, value):
```

Comment:
**Optional**
`path` instead of `value` here. 'Value' implies I can pass true/false to fill() to either check or uncheck (both being a 'fill' action).